src/profiles/profiles.controller.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  Controller,
+  Post,
+  Delete,
+  Param,
+  Body,
+  UseGuards,
+  Request,
+} from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import {
+  ApiBearerAuth,
+  ApiCreatedResponse,
+  ApiParam,
+  ApiTags,
+} from '@nestjs/swagger';
+
+import { Follow } from '@src/profiles/domain/profiles';
+
+import { FollowerActionDto } from './dto/follower.dto';
+import { ProfilesService } from './profiles.service';
+
+@ApiTags('Profiles')
+@Controller({
+  path: 'profiles',
+  version: '1',
+})
+export class ProfilesController {
+  constructor(private readonly followService: ProfilesService) {}
+
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard('jwt'))
+  @Post(':username/follow')
+  @ApiParam({
+    name: 'username',
+    type: String,
+    required: true,
+  })
+  @ApiCreatedResponse({
+    type: Follow,
+  })
+  async follow(
```

Comment:
Since await is not being used and the global ResolvePromisesInterceptor will handle promise resolution, you can remove the async keyword from the follow method. The method should directly return the promise from followService.follow.

Reply:
These changes you suggested are done please review it @dev-ahmadbilal